packages/hocs/withFocusTrap.jsx

```diff
@@ -1,27 +1,47 @@
 /* eslint-disable jsx-a11y/no-noninteractive-tabindex */
-import React, { useRef } from 'react'
+import React, { useRef, useEffect } from 'react'
+import PropTypes from 'prop-types'
 
 // Returns an array of focusable elements in the order they are found in c
-const getFocusable = c =>
-  c.querySelectorAll('button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])')
+const selector =
+  'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"]), audio[controls], video[controls], [contenteditable]:not([contenteditable=false]'
+const getFocusable = c => c.querySelectorAll(selector)
 
 const withFocusTrap = Component => {
   const WithFocusTrap = props => {
     const componentRef = useRef(null)
 
+    useEffect(() => {
+      // setting the focus to the first focusable element
+      const focusableElements = getFocusable(componentRef.current)[0]
+      if (focusableElements) focusableElements.focus()
+    })
+
+    // To force VoiceOver to treat the dialog as a modal we need to set the aria-label attribute.
+    // Also the modal-dialog html needs to be inserted into the page using JS after the page loads (this isn't a real problem)
+    const { ariaLabel, ...rest } = props
+
     const handleFocus = isEnd => () => {
       const focusableElements = getFocusable(componentRef.current)
       focusableElements[isEnd ? 0 : focusableElements.length - 1].focus()
     }
 
     return (
-      <div>
+      <div role="dialog" aria-modal="true" aria-label={ariaLabel}>
         <div onFocus={handleFocus(false)} tabIndex={0} />
-        <Component {...props} ref={componentRef} />
+        <Component {...rest} ref={componentRef} />
```

Comment:
This assumes the `Component` can accept a ref. Can you add a code comment that informs users that the `Component` must be able to accept a ref.